src/ValidationManifold.jl

```diff
@@ -122,55 +295,143 @@ function copyto!(
     X::ValidationFibreVector{TType};
     kwargs...,
 ) where {TType}
-    is_point(M, p; error = M.mode, kwargs...)
+    is_point(M, p; within = copyto!, context = (:Input,), kwargs...)
     copyto!(M.manifold, Y.value, p.value, X.value)
     return p
 end
 
+decorated_manifold(M::ValidationManifold) = M.manifold
+
 function distance(M::ValidationManifold, p, q; kwargs...)
-    is_point(M, p; error = M.mode, kwargs...)
-    is_point(M, q; error = M.mode, kwargs...)
-    return distance(M.manifold, array_value(p), array_value(q))
+    is_point(M, p; within = distance, context = (:Input,), kwargs...)
+    is_point(M, q; within = distance, context = (:Input,), kwargs...)
+    d = distance(M.manifold, internal_value(p), internal_value(q))
+    (d < 0) && _msg(
+        M,
+        DomainError(d, "Distance is negative.");
+        within = distance,
+        context = (:Output,),
+    )
+    return d
+end
+
+function embed(M::ValidationManifold, p; kwargs...)
+    is_point(M, p; within = embed, context = (:Input,), kwargs...)
+    q = embed(M.manifold, internal_value(p))
+    MEV = ValidationManifold(get_embedding(M.manifold), M)
+    is_point(MEV, q; error = MEV.mode, within = embed, context = (:Output,), kwargs...)
+    return ValidationMPoint(q)
+end
+function embed(M::ValidationManifold, p, X; kwargs...)
+    is_point(M, p; within = embed, context = (:Input,), kwargs...)
+    is_vector(M, p, X; within = embed, context = (:Input,), kwargs...)
+    Y = embed(M.manifold, internal_value(p), internal_value(X))
+    MEV = ValidationManifold(get_embedding(M.manifold), M)
+    is_vector(
+        MEV,
+        embed(M.manifold, internal_value(p)),
+        Y;
+        within = embed,
+        context = (:Output,),
+        kwargs...,
+    )
+    return ValidationTVector(Y)
```

Comment:
Oh, I have to check this carefully and add the base point (a copy of it) is active.